modules/couchbase/collect.go

```diff
@@ -0,0 +1,183 @@
+package couchbase
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/netdata/go.d.plugin/pkg/web"
+)
+
+const (
+	urlPathBucketsStats = "/pools/default/buckets"
+)
+
+func (cb *Couchbase) collect() (map[string]int64, error) {
+
+	ms := cb.scrapeCouchbase()
+	if ms.empty() {
+		return nil, nil
+	}
+	collected := make(map[string]int64)
+	err := cb.collectBasicStats(collected, ms)
+	if err != nil {
+		return nil, fmt.Errorf("error on creating a connection: %v", err)
+	}
+
+	return collected, nil
+}
+
+func (cb Couchbase) collectBasicStats(collected map[string]int64, ms *cbMetrics) error {
+	for _, b := range ms.BucketsStats {
+
+		if !cb.collectedBuckets[b.Name] && ms.hasBucketsStats() {
```

Comment:
> ms.hasBucketsStats()

is not needed, b/c we `range ms.BucketsStats`, there is no loop if len(ms.BucketsStats) < 1